pkg/operator/observe_config.go

```diff
@@ -157,6 +166,42 @@ func observeInternalRegistryHostname(listers Listers, observedConfig map[string]
 	return observedConfig, nil
 }
 
+func observeRoutingSubdomain(listers Listers, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	// TODO: Get the value for routingConfig.subdomain from global config or
+	// from a ClusterIngress resource instead of from the install config.
+	clusterConfig, err := listers.clusterConfigLister.ConfigMaps(clusterConfigNamespaceName).Get(clusterConfigName)
+	if errors.IsNotFound(err) {
+		return observedConfig, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if clusterConfig != nil {
+		if installConfig, ok := clusterConfig.Data["install-config"]; ok {
+			type InstallConfigMetadata struct {
+				Name string `json:"name"`
+			}
+
+			type InstallConfig struct {
+				Metadata   InstallConfigMetadata `json:"metadata"`
+				BaseDomain string                `json:"baseDomain"`
+			}
+
+			var config InstallConfig
+			if err := yaml.Unmarshal([]byte(installConfig), &config); err != nil {
+				return nil, fmt.Errorf("invalid InstallConfig: %v\njson:\n%s", err, installConfig)
+			}
+
+			subdomain := fmt.Sprintf("apps.%s.%s", config.Metadata.Name, config.BaseDomain)
```

Comment:
`apps` is hardcoded?  that surprises me.

Reply:
Really? 😁 

We have outstanding work to pull this out to cluster config one way or another.